src/storage/mod.rs

```diff
@@ -653,6 +655,13 @@ impl Storage {
         options: Options,
         callback: Callback<Vec<Result<()>>>,
     ) -> Result<()> {
+        for m in &mutations {
+            let (size, limit) = (m.key().encoded().len(), self.max_key_size);
```

Comment:
seem needless to define the limit variable.

Reply:
`Error::KeyTooLarge { size, self.max_key_size }` will not compile, rustc complains something about 'expect self.max_key_size to be an identifier' :(